solutions/usaco-809.mdx

```diff
@@ -0,0 +1,66 @@
+---
+id: usaco-809
+source: USACO Bronze 2018 Febuary
+title: Taming the Herd
+author: Sathvik Chundru
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_bronze_feb18.html)
+
+<LanguageSection>
+
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+class Main {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("taming");
+		int n = io.nextInt();
+		int[] log = new int[100000];
+		for(int i = 0; i < n; i++){
+			log[i] = io.nextInt();
+		}
+		if(log[0] > 0){
+			io.println(-1);
+			io.close();
+			System.exit(0);
+		}
+		log[0] = 0;
+		int t = -1;
+		int req = 0;
```

Comment:
comments would be nice

Reply:
the official cpp solution explains it better in paragraph form so no need to make code ugly by adding comments that will never do as good a job of explaining the problem as the official solution. Mine is just for people who do java so they can understand what is going on in the cpp code.

Reply:
...then copy paste the comments?

Reply:
its like 3 paragraphs above the code. If you want me to I can include it in mine

Reply:
if you mean it's included in the editorial, i'd suggest you include the relevant parts of it in this code so people know which code segment matches to which editorial sentence
(it'd be cool if you could do this for the cpp version as well)
but if there's a editorial i wouldn't really care if you included it or not

Reply:
I will try later

Reply:
fixed @HarryWangATX @dongliu0426 if its good merge now

Reply:
@SansPapyrus683